File: Programming_And_Backend_Development/Exercises/RB101_RB119_Small_Problems/Easy_9/welcomestranger.rb
=begin
Create a method that takes 2 arguments, an array and a hash. The array will
contain 2 or more elements that, when combined with adjoining spaces, will
produce a person's name. The hash will contain two keys, :title and :occupation,
and the appropriate values. Your method should return a greeting that uses the
person's full name, and mentions the person's title and occupation.

Example:
greetings(['John', 'Q', 'Doe'], { title: 'Master', occupation: 'Plumber' })
=> "Hello, John Q Doe! Nice to have a Master Plumber around."

=end

def greet (array, hash)
  name = array.join(" ")
  job = "#{hash[:title]} #{hash[:occupation]}"
  puts "Hello, #{name}. You are a #{job}."
end


name = ["Pedro", "Lima"]
job = {title: "Senior", occupation: "Mechanic"}

greet(name, job)

File: front_end_development/JS210/welcome_stranger.js
// Write a function that takes two arguments, an array and an object. The
// array will contain two or more elements that, when combined with
// spaces, produce a person's full name. The object will contain two keys,
// title and occupation, and suitable values for both items. Your function
// should log a greeting to the console that uses the person's full name,
// and mentions the person's title and occupation.

// greetings(['John', 'Q', 'Doe'], { title: 'Master', occupation: 'Plumber' });

// Hello, John Q Doe! Nice to have a Master Plumber around.

function greetings(fullName, jobDescription) {
  let message = ['Hello, ', '', '! Nice to have a ', '', ' around.'];
  message[1] = fullName.join(' ');
  message[3]= (jobDescription.title + ' ' + jobDescription.occupation);
  console.log(message.join(''));
}

greetings(['Pedro', 'Lima'], { title: 'Senior', occupation: 'Mechanic'});
